client/frame-runner.js

```diff
@@ -37,32 +38,68 @@ document.addEventListener('DOMContentLoaded', function() {
       // add delay here for firefox to catch up
       .delay(200)
       /* eslint-disable no-unused-vars */
-      .map(({ text, testString }) => {
+      .flatMap(({ text, testString }) => {
         const assert = chai.assert;
       /* eslint-enable no-unused-vars */
         const newTest = { text, testString };
         let test;
+        let __result;
         try {
           /* eslint-disable no-eval */
+          // eval test string to actual JavaScript
+          // This return can be a function
+          // i.e. function() { assert(true, 'happy coding'); }
           test = eval(testString);
           /* eslint-enable no-eval */
           if (typeof test === 'function') {
-            // maybe sync/promise/observable
+
+            // we know that the test eval'ed to a function
+            // the function could expect a callback
+            // or it could return a promise/observable
+            // or it could still be sync
             if (test.length === 0) {
-              test();
+              // a function with length 0 means it expects 0 args
+              // We call it and store the result
+              // This result may be a promise or an observable or undefined
+              __result = test();
+            } else {
+              // if function takes arguments
+              // we expect it to be of the form
+              // function(cb) { /* ... */ }
+              // and callback has the following signature
+              // function(err) { /* ... */ }
+              __result = Rx.Observable.fromNodeCallback(test);
```

Comment:
I'm a rx newbie, but here I think you need this : 
` __result = Rx.Observable.fromNodeCallback(test)();`
I wasn't able to make the callback approach work, then I used the debugger and I noticed that `fromNodeCallback` returns a function, and this function returns the `Observable`.
Adding the `()` function call the callback approach works correctly